backend/src/test/java/fk/prof/backend/LeaderElectionTest.java

```diff
@@ -0,0 +1,244 @@
+package fk.prof.backend;
+
+import fk.prof.backend.service.ProfileWorkService;
+import fk.prof.backend.util.IPAddressUtil;
+import fk.prof.backend.model.election.LeaderDiscoveryStore;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.RetryOneTime;
+import org.apache.curator.test.TestingServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+@RunWith(VertxUnitRunner.class)
+public class LeaderElectionTest {
+
+  private Vertx vertx;
+  private Integer leaderPort;
+  private JsonObject leaderHttpServerConfig;
+  private JsonObject config;
+  private JsonObject vertxConfig;
+  private DeploymentOptions leaderElectionDeploymentOptions;
+
+  private TestingServer testingServer;
+  private CuratorFramework curatorClient;
+
+  @Before
+  public void setUp(TestContext context) throws Exception {
+    ConfigManager.setDefaultSystemProperties();
+    config = ConfigManager.loadFileAsJson(LeaderElectionTest.class.getClassLoader().getResource("config.json").getFile());
+    vertxConfig = ConfigManager.getVertxConfig(config);
+
+    JsonObject leaderElectionDeploymentConfig = ConfigManager.getLeaderElectionDeploymentConfig(config);
+    assert leaderElectionDeploymentConfig != null;
+
+    leaderHttpServerConfig = ConfigManager.getLeaderHttpServerConfig(config);
+    assert leaderHttpServerConfig != null;
+    leaderPort = leaderHttpServerConfig.getInteger("port");
+    leaderElectionDeploymentOptions = new DeploymentOptions(leaderElectionDeploymentConfig);
+
+    testingServer = new TestingServer();
+    curatorClient = CuratorFrameworkFactory.newClient(testingServer.getConnectString(), 500, 500, new RetryOneTime(1));
+    curatorClient.start();
+    curatorClient.blockUntilConnected(10, TimeUnit.SECONDS);
+  }
+
+  @After
+  public void tearDown(TestContext context) throws IOException {
+    System.out.println("Tearing down");
+    VertxManager.close(vertx).setHandler(result -> {
+      System.out.println("Vertx shutdown");
+      curatorClient.close();
+      try {
+        testingServer.close();
+      } catch (IOException ex) {
+      }
+      if (result.failed()) {
+        context.fail(result.cause());
+      }
+    });
+  }
+
+  @Test(timeout = 20000)
+  public void leaderTaskTriggerOnLeaderElection(TestContext testContext) throws InterruptedException {
```

Comment:
I see no assert here, is the test going to timeout and that will be the failure? If so its worth commenting in code.

Usually, asserts tells the reader what invariants the test cares about, no-assert implies no-invariants.